golint/golint.go

```diff
@@ -42,23 +42,43 @@ func main() {
 	switch flag.NArg() {
 	case 0:
 		lintDir(".")
-	case 1:
-		arg := flag.Arg(0)
-		if strings.HasSuffix(arg, "/...") && isDir(arg[:len(arg)-4]) {
-			for _, dirname := range allPackagesInFS(arg) {
-				lintDir(dirname)
+	default:
+		// dirsRun and filesRun indicate whether the golint is applied to directory
+		// or file targets. The distinction affects which checks are run. If both
+		// are false, the golint is appplied to package targets.
+		var dirsRun, filesRun bool
```

Comment:
You need another flag for `pkgsRun`. Otherwise it is possible to mix files and packages and your check below will not catch it.